compiler/rustc_mir_transform/src/pass_manager.rs

```diff
@@ -77,17 +77,30 @@ pub fn run_passes<'tcx>(tcx: TyCtxt<'tcx>, body: &mut Body<'tcx>, passes: &[&dyn
     let mut cnt = 0;
 
     let validate = tcx.sess.opts.debugging_opts.validate_mir;
+    let overridden_passes = &tcx.sess.opts.debugging_opts.mir_enable_passes;
+    trace!("Overridden: {:?}", overridden_passes);
 
     if validate {
         validate_body(tcx, body, format!("start of phase transition from {:?}", start_phase));
     }
 
     for pass in passes {
-        if !pass.is_enabled(&tcx.sess) {
-            continue;
-        }
-
         let name = pass.name();
+
+        if let Some((_, polarity)) = overridden_passes.iter().rev().find(|(s, _)| s == &*name) {
```

Comment:
I had considered this but actually intentionally avoided it - my reasoning was that the most important scenario for perf is the one where `overridden_passes` is empty (since that's what essentially every user will be using), and `Vec` is a data structure that I can be very certain is very fast in that case.